driver-sync/src/test/functional/com/mongodb/client/AbstractUnifiedTest.java

```diff
@@ -116,8 +119,14 @@
 
     private static final long MIN_HEARTBEAT_FREQUENCY_MS = 50L;
 
+    /**
+     * @param createSessions {@code true} means that {@code session0}, {@code session1} {@link ClientSession}s must be created as specified
```

Comment:
Had to introduce which switch to avoid a race condition in `minPoolSize-error.json`. The problem was discussed in Slack (https://mongodb.slack.com/archives/G016S9LLKU2/p1618781972024600). As an example, Rust does not have the problem because there creating a session object does not result in requesting `startSession`.

Reply:
Added a small code comment and a detailed commit message here d94b2e29d0ed831fa8975d110ae326a865d13db9.

Reply:
I think the comment is super helpful for future maintainers :+1: